src/store.js

```diff
@@ -272,13 +277,48 @@ export function setActiveAndEntryKeyStates(version, context, config, configKeySt
   });
 }
 
+export function evaluateAllocationPredicates(context, allocation, activeKeyStates) {
```

Comment:
This function reads in an `allocation`, along with the current `context` and `activeKeyStates`. Using the currently active keys housed in `activeKeyStates`, this function traverses the allocations genome to determine if any predicated variant exist within.

Once a predicated variant is found it uses the current context to evaluate, in order, the list of possible variants, matching the variants predicate with the current context. The first variant that matches is chosen, if no variants match from the `_predicated_values` array, the `_default_value` is chosen. 

The key of the matching variant is stored in the active keys set to be used to activate the css / js in the assets files.